worker/jobs/bulkUpsertSamplesJob.js

```diff
@@ -54,22 +54,28 @@ module.exports = (job, done) => {
         if (result.isFailed) {
           errors.push(result.explanation);
           return Promise.resolve();
-        } else {
-          successCount++;
-          if (featureToggles.isFeatureEnabled('publishSampleInPromiseChain')) {
-            // Wait for publish to complete before resolving the promise.
-            return publisher.publishSample(result, subHelper.model);
-          }
+        }
 
-          /*
-           * Resolve the promise right away, *before* we actually publish
-           * the sample. Under heavy load, publish a sample can get backed up
-           * because we are looking up the subject info and including that in
-           * the payload of the real-time event.
-           */
-          publisher.publishSample(result, subHelper.model);
-          return Promise.resolve();
+        successCount++;
+        if (featureToggles.isFeatureEnabled('publishSampleInPromiseChain')) {
+          // Wait for publish to complete before resolving the promise.
+          return publisher.publishSample(result, subHelper.model)
+            .catch((err) => {
```

Comment:
Since both code paths calling publisher.publishSample are just catching and logging, maybe move the catch and resolve to the promise which publishSample is returning instead?